File: programmers/ForKaKAO/2018/level1/비밀지도.cpp
#include <string>
#include <vector>
#include <bitset>
#include <algorithm>
#include <iostream>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    
    for (int i = 0; i < n; i++) {
        int row = arr1[i] | arr2[i];
        
        string str = "";
        while(str.length() != n) {
            if (row % 2 == 1) str += "#";
            else str += " ";
            row /= 2;
        }
        reverse(str.begin(), str.end());
        answer.push_back(str);
    }
    return answer;
}

File: programmers/ForKaKAO/2018/level1/비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for num1, num2 in zip(arr1, arr2):
        sum = str(bin(num1 | num2)[2:])
        sum = sum.rjust(n, '0')
        sum = sum.replace('1', '#')
        sum = sum.replace('0', ' ')
        answer.append(sum)
    return answer